sources/SessionTracker/SessionHandler.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+namespace Combodo\iTop\SessionTracker;
+
+use Combodo\iTop\Application\Helper\Session;
+use ContextTag;
+use Exception;
+use IssueLog;
+use UserRights;
+use utils;
+
+/**
+ * Class SessionHandler
+ *
+ * @author Olivier Dain <olivier.dain@combodo.com>
+ * @package Combodo\iTop\SessionTracker
+ * @since 3.1.1 3.2.0 N°6901
+ */
+class SessionHandler extends \SessionHandler
+{
+	public function destroy($session_id)
+	{
+		IssueLog::Debug("destroy", \LogChannels::SESSIONTRACKER, [
+			'session_id' => $session_id,
+		]);
+		$bRes = parent::destroy($session_id);
+
+		if ($bRes) {
+			$this->unlink_session_file($session_id);
+		}
+
+		return $bRes;
+	}
+
+	public function gc($max_lifetime)
+	{
+		IssueLog::Debug("gc", \LogChannels::SESSIONTRACKER, [
+			'max_lifetime' => $max_lifetime,
+		]);
+		$iRes = parent::gc($max_lifetime);
+		$this->gc_with_time_limit($max_lifetime);
+		return $iRes;
+	}
+
+	public function open($save_path, $session_name)
+	{
+		$bRes = parent::open($save_path, $session_name);
+
+		$session_id = session_id();
+		IssueLog::Debug("open", \LogChannels::SESSIONTRACKER, [
+			'session_id' => $session_id,
+		]);
+
+		if ($bRes) {
+			$this->touch_session_file($session_id);
+		}
+
+		return $bRes;
+	}
+
+	public function write($session_id, $data)
+	{
+		$bRes = parent::write($session_id, $data);
+
+		IssueLog::Debug("write", \LogChannels::SESSIONTRACKER, [
+			'session_id' => $session_id,
+			'data' => $data,
+		]);
+
+		if ($bRes) {
+			$this->touch_session_file($session_id);
+		}
+
+		return $bRes;
+	}
+
+	public static function session_set_save_handler() : void
+	{
+		session_set_save_handler(new SessionHandler(), true);
+	}
+
+	private function generate_session_content(?string $sPreviousFileVersionContent) : ?string
+	{
+		try {
+			$sUserId = UserRights::GetUserId();
+			if (is_null($sUserId)) {
+				return null;
+			}
+
+			// Default value in case of
+			// - First time file creation
+			// - Data corruption (not a json / not an array / no previous creation_time key)
+			$iCreationTime = time();
+
+			if (! is_null($sPreviousFileVersionContent)) {
+				$aJson = json_decode($sPreviousFileVersionContent, true);
+				if (is_array($aJson) && array_key_exists('creation_time', $aJson)) {
+					$iCreationTime = $aJson['creation_time'];
+				}
+			}
+
+			return json_encode (
+				[
+					'login_mode' => Session::Get('login_mode'),
+					'user_id' => $sUserId,
+					'creation_time' => $iCreationTime,
+					'context' => implode("|", ContextTag::GetStack())
```

Comment:
```suggestion
					'contexts' => implode("|", ContextTag::GetStack())
```

Making this plural as it contains all context tags currently set.
@odain-cbd you might need to adjust your monitoring module.